particle-iot/particle-cli
# bugfix/ch47215/product-error-messaging

## Description

Improves error messaging when `--product <id>` flag is set but `device` param is not an ID ([ch](https://app.clubhouse.io/particle/story/47215/unhelpful-error-message-when-using-device-name-and-product-flag))

## How to Test

_Note: requires you to have a Particle product with one or more devices_

Run the following commands and note the error message you see:

`particle cloud flash <device-name> <path-to-project.ino> --product <product-id>`
`particle variable get <device-name> <var-name> --product <product-id>`
`particle subscribe --device <device-name> --product <product-id>`
`particle function call <device-name> <fn-name> --product <product-id>`

Then run `particle --help` and confirm all functions which currently support the `--product` flag as well as the `product` commands themselves properly handle being called with a `<device-name>`

## Completeness
